src/base_model.jl

```diff
@@ -275,16 +276,27 @@ function build_base_model!(ECModel::AbstractEC, optimizer; use_notations=false)
         con_us_bat_balance[u in user_set, b in asset_names(users_data[u], BATT), t in time_set],
         #E_batt_us[u, b, t] - E_batt_us[u, b, if (t>1) t-1 else final_step end]  # Difference between the energy level in the battery. Note that in the case of the first time step, the last id is used
         E_batt_us[u, b, t] - E_batt_us[u, b, pre(t, time_set)]  # Difference between the energy level in the battery. Note that in the case of the first time step, the last id is used
-        + profile(market_data, "time_res")[t] * P_conv_P_us[u, field_component(users_data[u], b, "corr_asset"), t]/(
+        + profile(gen_data,"time_res")[t] * P_conv_P_us[u, field_component(users_data[u], b, "corr_asset"), t]/(
             sqrt(field_component(users_data[u], b, "eta"))*field_component(users_data[u], field_component(users_data[u], b, "corr_asset"), "eta"))  # Contribution of the converter when supplying power to AC
-        - profile(market_data, "time_res")[t] * P_conv_N_us[u, field_component(users_data[u], b, "corr_asset"), t]*(
+        - profile(gen_data,"time_res")[t] * P_conv_N_us[u, field_component(users_data[u], b, "corr_asset"), t]*(
             sqrt(field_component(users_data[u], b, "eta"))*field_component(users_data[u], field_component(users_data[u], b, "corr_asset"), "eta"))  # Contribution of the converter when absorbing power from AC
         == 0
     )
     
     return ECModel
 end
 
+"""
+#This function allow to get the profile of each users related to their market type (e.g. commercial, non_commercial)

```

Comment:
Super!
The convention here is that you shall replicate the example of calculate demand:
The first line contains a tab and the definition of the function, in this case, it should be: [TAB]market_profile_by_user(ECModel::AbstractEC, u_name, profile_name)

You don't need the # inside the code.
Consider that the """{Any multiline string}""" is used to create a multiline string. Actually you can use that anywhere in the text.

Then, you add an empty line and after that the description of the function.
In some cases, for relevant functions, we may also add Inputs and Outputs You may seen the corresponding convention in the other function.

It may be interesting to add inputs and outputs also here.